internal/grpc/services/publicstorageprovider/publicstorageprovider.go

```diff
@@ -458,9 +462,22 @@ func (s *service) Stat(ctx context.Context, req *provider.StatRequest) (*provide
 			Value: attribute.StringValue(req.Ref.String()),
 		})
 
-	tkn, relativePath, err := s.unwrap(ctx, req.Ref)
-	if err != nil {
-		return nil, err
+	var (
+		tkn          string
+		relativePath string
+		nodeID       string
+	)
+
+	if req.Ref.ResourceId != nil {
+		parts := strings.Split(req.Ref.ResourceId.OpaqueId, "/")
+		tkn = parts[0]
+		nodeID = parts[1]
+	} else if req.Ref.Path != "" {
+		var err error
+		tkn, relativePath, err = s.unwrap(ctx, req.Ref)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	originalPath, ls, ri, st, err := s.resolveToken(ctx, tkn)
```

Comment:
If `ri` is a file, then `relativePath` should be empty, right? We can save the extra stat call in that case and return early

Similarly, in `resolveToken`, we call `GetPath` and `Stat` but we can just return the path returned by the stat response